lib/xdparser2.ts

```diff
@@ -387,33 +387,34 @@ export function xdParser(xd: string, strict = false, editorInfo = false): Crossw
   }
 }
 
-function replaceWordWithSymbol(word: string, tiles: Tile[], splitChar: string) {
+export function replaceWordWithSymbol(word: string, tiles: Tile[], splitChar: string) {
   let newWord = ""
 
   let tileIdx = 0
-  let offset = 0
-  for (let i = 0; i < word.length; i++) {
-    const cur = word[i + offset]
-
-    // don't increment tileIdx when adding splits
-    if (cur === splitChar) {
-      newWord += cur
-      continue
-    }
+  for (let i = 0; i < word.length && tileIdx < tiles.length;) {
```

Comment:
I don't know if I've ever seen a for loop written like this. A while loop might read a little clearer - not a blocker.